pymc/logprob/utils.py

```diff
@@ -201,18 +203,120 @@ def __str__(self):
         return f"Check{{{self.msg}}}"
 
 
-@node_rewriter(tracks=[CheckParameterValue])
+class InnerGraphRewriter:
```

Comment:
This may be overdoing it, but what do you think of implementing this as something like WalkingGraphRewriter: https://github.com/ricardoV94/pytensor/blob/9cf2d181f07dc99bbd2e7c9e2b4a3e1b0aeff034/pytensor/graph/rewriting/basic.py#L1998

A `WalkingNestedGraphRewriter` which applies the same `node_rewriter` to both the outer graph and inner graphs. The idea is you would pass the previous rewrite which doesn't distinguish between the core case or a `Scan` `OpFromGraph`. It would be the `WalkingNestedGraphRewriter` that would apply that logic regardless of what `NodeRewriter` it's given?

Reply:
Hmm, sounds great. I've been thinking about something like that, not sure how to best implement it.